### PR TITLE
feat:  添加语言配置

### PR DESCRIPTION
**Pull request by LanjianNUll**

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/vConsole?pullRequest=593) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/vConsole?pullRequest=593) before we can accept your contribution.<br/><hr/>**lanjian** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/vConsole?pullRequest=593) it.</sub>

File: src/core/core.svelte [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <script lang="ts">
   import { onMount, onDestroy, createEventDispatcher } from 'svelte';
   import * as tool from '../lib/tool';
+  import { i18n } from '../lib/lang'
```
Comment (ManiaciaChao) [inline src/core/core.svelte:4]: need format

File: src/core/core.svelte [hunk 2]
```diff
@@ -66,6 +70,7 @@
         showMask = false;
       }, 330);
     }
+    t = i18n(lang)
```
Comment (ManiaciaChao) [inline src/core/core.svelte:73]: semicolon

File: src/core/core.svelte [hunk 3]
```diff
@@ -340,7 +345,7 @@
             class:vc-actived={plugin.id === activedPluginId}
             id="__vc_tab_{plugin.id}"
             on:click={() => onTapTabbar(plugin.id)}
-          >{plugin.name}</a>
+          >{ t(plugin.name)}</a>
```
Comment (ManiaciaChao) [inline src/core/core.svelte:348]: redundant space

File: src/core/core.svelte [hunk 4]
```diff
@@ -385,10 +390,10 @@
             class:vc-global-tool={item.global}
             class:vc-toggle={plugin.id === activedPluginId}
             on:click={(e) => onTapToolbar(e, plugin.id, i)}
-          >{item.name}</i>
+          >{t(item.name)}</i>
         {/each}
       {/each}
-      <i class="vc-tool vc-global-tool vc-tool-last vc-hide" on:click={onTapEventHide}>Hide</i>
+      <i class="vc-tool vc-global-tool vc-tool-last vc-hide" on:click={onTapEventHide}>{t("Hide")}</i>
```
Comment (ManiaciaChao) [inline src/core/core.svelte:396]: single quotes

File: src/core/core.ts [hunk 5]
```diff
@@ -232,6 +232,8 @@ export class VConsole {
       this.compInstance.theme = theme;
     }
 
+    this.compInstance.lang = this.option.lang ? this.option.lang : "en"
```
Comment (ManiaciaChao) [inline src/core/core.ts:235]: ` this.compInstance.lang = this.option.lang ?? 'en';`

File: src/lib/lang.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+// https://webpack.js.org/guides/dependency-management/#requirecontext
```
Comment (ManiaciaChao) [inline src/lib/lang.ts:1]: this whole file need to be reformated :)

**Review by ManiaciaChao — CHANGES_REQUESTED** [review]
Since this project hasn't been configured with CI, I have to manually do a coding style check.
Basically, LGTM.  Thanks for reaching out!

File: src/core/core.svelte (hunk 1 above)
Comment (LanjianNUll) [reply src/core/core.svelte:4]: semicolon？

File: src/core/core.svelte [hunk 7]
```diff
@@ -44,6 +47,7 @@
   let preivousContentUpdateTime = 0;
   let cssTimer = null;
   const contentScrollTop: { [pluginId: string]: number } = {};
+  let t = i18n(lang)
```
[the remark this answers is not in the document]
Comment (LanjianNUll) [reply src/core/core.svelte:50]: ok

File: src/core/core.svelte (hunk 2 above)
Comment (LanjianNUll) [reply src/core/core.svelte:73]: ok

File: src/core/core.svelte (hunk 3 above)
Comment (LanjianNUll) [reply src/core/core.svelte:348]: ok

File: src/core/core.svelte (hunk 4 above)
Comment (LanjianNUll) [reply src/core/core.svelte:396]: ok

File: src/core/core.ts (hunk 5 above)
Comment (LanjianNUll) [reply src/core/core.ts:235]: yes

**Comment by LanjianNUll** [comment]
Thanks Reply.Recently, I have been thinking about how to dynamically change the language, but I still haven't solved it. I always feel that this submission is not perfect

**Comment by ManiaciaChao** [comment]
> Thanks Reply.Recently, I have been thinking about how to dynamically change the language, but I still haven't solved it. I always feel that this submission is not perfect

For Svelte application, you should use [stores](https://svelte.dev/tutorial/writable-stores) to dynamically change strings.
There is already a nice implemetation https://www.npmjs.com/package/svelte-i18n.

**Comment by Maizify** [comment]
Thanks for submitting! vConsole is a lightweight tool, and it is geared towards developers. I think the words in it are easy to understand. Considering performance, I prefer to keep it as it is. If you have to do this, I also recommend considering mature solutions such as svelte-i18n.

**Pull request closed without merging**